### ISSUE TITLE
Log-off button not working with SimpleSAMLphp?

### ISSUE THREAD

**Issue by kokarl** [filesender/filesender#1777]
On a new fresh install from sources : 
- filesender 2.44
- SimpleSAMLphp 2.1.1
- Apache 2.4.58
- PHP 8.1

Everything is working fine except when a user press the log-off button on the menu. He is redirected on a SimpleSAMLphp Unhandled exception web page with the following error message :  

SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
2 src/SimpleSAML/Error/ExceptionHandler.php:32 (SimpleSAML\Error\ExceptionHandler::customExceptionHandler)
1 vendor/symfony/error-handler/ErrorHandler.php:541 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
0 [builtin] (N/A)
Caused by: Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException: No route found for "POST https://[myfilesender.url.com]/simplesaml/module.php/core/logout/example-userpass": Method Not Allowed (Allow: GET)
Backtrace:
7 vendor/symfony/http-kernel/EventListener/RouterListener.php:134 (Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelRequest)
6 vendor/symfony/event-dispatcher/EventDispatcher.php:230 (Symfony\Component\EventDispatcher\EventDispatcher::callListeners)
5 vendor/symfony/event-dispatcher/EventDispatcher.php:59 (Symfony\Component\EventDispatcher\EventDispatcher::dispatch)
4 vendor/symfony/http-kernel/HttpKernel.php:139 (Symfony\Component\HttpKernel\HttpKernel::handleRaw)
3 vendor/symfony/http-kernel/HttpKernel.php:75 (Symfony\Component\HttpKernel\HttpKernel::handle)
2 vendor/symfony/http-kernel/Kernel.php:202 (Symfony\Component\HttpKernel\Kernel::handle)
1 src/SimpleSAML/Module.php:234 (SimpleSAML\Module::process)
0 public/module.php:17 (N/A)

The URL displayed in the web browser is : 
https://myfilesender.url.com/simplesaml/module.php/core/logout/example-userpass?ReturnTo=https%3A%2F%2Fmyfilesender.url.com%2F%3Fs%3DlogoutAuthId=example-userpass&ReturnTo=https%3A%2F%2Fmyfilesender.url.com%2F%3Fs%3Dlogout

The URL seems to be correct, because when the user validate the previous URL by pressing enter in its browser, he is correctly disconnected and redirected to https://myfilesender.url.com/?s=logout

**Comment by WebSpider** [comment]
This seems to be the problem:

```
MethodNotAllowedHttpException: No route found for "POST https://[myfilesender.url.com]/simplesaml/module.php/core/logout/example-userpass": Method Not Allowed (Allow: GET)
```
You may need to configure either side to agree on the correct HTTP method to use

**Comment by kokarl** [comment]
Yes, I tried to modify the template/menu.php file.
The link for the log-off button is : "<form action="$link" method="post">"
I changed by "<form action="$link" method="get">". The user is well logged off, but he is redirected to the SimpleSAMLphp Welcome page and not to https://myfilesender.url.com/?s=logout.

**Comment by monkeyiq** [comment]
I see also if I change my post to get then I will be logged out but also not redirected back. I am not sure why your POST is being denied by SSP/Symfony. I've been trying to see where that is defined in SSP.

**Comment by monkeyiq** [comment]
If you change `modules/core/routing/routes/routes.yml` adding POST to the following block the normal logout should work for you in SimpleSAMLphp 2.1.1.

```
...
core-logout:
  path: /logout/{as}
  defaults: {
    _controller: 'SimpleSAML\Module\core\Controller\Logout::logout'
  }
  methods: [GET,POST]
...
```

**Comment by monkeyiq** [comment]
This is an interesting conflict. A security audit on FileSender wanted the logout to be only performed using POST and the SSP has updated to explicitly forbid it.

**Comment by kokarl** [comment]
Many thanks @monkeyiq  !! It really works well now. It would be worth asking SSP dev team why they forbid POST now...

**Comment by WebSpider** [comment]
Had a quick look, and this seems related to https://github.com/simplesamlphp/simplesamlphp/commit/b342e262acbc2e3cc7e8eea04545d0af5fa644c5, which introduces methods on all used routes.

**Comment by monkeyiq** [comment]
I have asked about this on the SSP dev google group.

**Comment by monkeyiq** [comment]
After a chat with the SSP team I have committed https://github.com/filesender/filesender/pull/1783 to address this issue

**Comment by monkeyiq** [comment]
This update will be part of release 2.45

**Comment by monkeyiq** [comment]
https://github.com/filesender/filesender/releases/tag/filesender-2.45

**Comment by monkeyiq** [comment]
Thank you for the report. This should be resolved in release 2.45.
